shared/stateutil/arrays.go

```diff
@@ -64,33 +62,25 @@ func (h *stateRootHasher) arraysRoot(roots [][]byte, fieldName string) ([32]byte
 		if maxChangedIndex+2 == len(chunks) && maxChangedIndex%2 != 0 {
 			changedIndices = append(changedIndices, maxChangedIndex+1)
 		}
-
 		for i := 0; i < len(changedIndices); i++ {
 			rt, err = recomputeRoot(changedIndices[i], chunks, fieldName)
 			if err != nil {
 				return [32]byte{}, err
 			}
 		}
+		lock.Lock()
+		leavesCache[fieldName] = chunks
+		lock.Unlock()
 		return rt, nil
 	}
 
-	hashKey := hashutil.FastSum256(hashKeyElements)
```

Comment:
This cache will never get hit. Arrays always typically get one element added each slot.